misk/src/main/kotlin/misk/client/HttpClientsConfig.kt

```diff
@@ -1,115 +1,75 @@
 package misk.client
 
+import com.fasterxml.jackson.annotation.JsonAlias
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize
-import com.fasterxml.jackson.databind.type.TypeFactory
 import misk.config.Config
+import misk.logging.getLogger
 import misk.security.ssl.CertStoreConfig
 import misk.security.ssl.TrustStoreConfig
 import java.io.File
+import java.net.URL
 import java.time.Duration
 
-data class BackwardsCompatibleEndpointConfig(
```

Comment:
Extracted most of "backwards compatibility" stuff into separate file, to make it less messy.